grails-app/controllers/kuorum/politician/CampaignController.groovy

```diff
@@ -527,14 +527,9 @@ class CampaignController {
 
 
     def validateUserPhoneSendSMS(DomainUserPhoneValidationCommand domainUserPhoneValidationCommand) {
-        def isCaptchaVerified = false
-        def isCaptchaNecessary = Boolean.parseBoolean(params['allowAnonymousAction']);
-        if (isCaptchaNecessary) {
-            isCaptchaVerified = captchaService.verifyCaptcha(params['g-recaptcha-response']);
-        } else {
-            isCaptchaVerified = true;
-        }
-        if (domainUserPhoneValidationCommand.hasErrors() || !isCaptchaVerified) {
+        def captchaPassed = false;
+        captchaPassed = springSecurityService.isLoggedIn() || captchaService.verifyCaptcha(params['g-recaptcha-response']);
```

Comment:
In only one line:
`boolean captchaPassed = springSecurityService.isLoggedIn() || captchaService.verifyCaptcha(params['g-recaptcha-response']);
`